### PR TITLE
Fix coarsen specification in non-conservative VC integrator

### PR DESCRIPTION
**Pull request by nishantnangia**
This PR fixes and implements tests for the issue described by https://github.com/IBAMR/IBAMR/issues/865

### CODE REVIEW

File: tests/vc_navier_stokes/SetFluidProperties.cpp [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+//
+// This file is part of IBAMR.
+//
+// IBAMR is free software and is distributed under the 3-clause BSD
+// license. The full text of the license can be found in the file
+// COPYRIGHT at the top level directory of IBAMR.
+//
+// ---------------------------------------------------------------------
+
+// Filename: SetFluidProperties.cpp
```
Comment (boyceg) [inline tests/vc_navier_stokes/SetFluidProperties.cpp]: @drwells, I think we are omitting this material from source code files now --- is that right?

Comment (drwells) [reply tests/vc_navier_stokes/SetFluidProperties.cpp]: Yes.

File: tests/vc_navier_stokes/SetFluidProperties.h [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE
```
Comment (drwells) [inline tests/vc_navier_stokes/SetFluidProperties.h]: same here - we just need the new and equivalent text.

**Comment by drwells** [comment]
Thanks a lot for the fix! Would you please also add a changelog entry so you get credit for it at release time?

**Comment by nishantnangia** [comment]
I trimmed the license stuff, added the changelog, and squashed.

**Comment by drwells** [comment]
I can confirm that the output from these tests matches the output from versions of IBAMR before the bug was introduced, so lets merge!

**Pull request merged**